config/Makefile.mingw

```diff
@@ -100,6 +100,12 @@ BYTECC=$(TOOLPREF)gcc
 ### Additional compile-time options for $(BYTECC).  (For static linking.)
 BYTECCCOMPOPTS=-O -mms-bitfields -Wall -Wno-unused
 
+### Additional compile-time options for $(BYTECC).  (For debug version.)
+BYTECCDBGCOMPOPTS=-g
+
+### Flag to use to rename object files.  (for debug version.)
+NAME_OBJ_FLAG=-o
```

Comment:
You're right, gcc accepts `-omyprogram.exe`, so maybe Damien's space dance is not necessary.  

Another way to deal with those space-or-not-space issues is to define macros that take parameters and are to be invoked via `$(call...)`, like `MKLIB` or `MAKE_OCAMLRUN`.